thunder/core/vjp_utils.py

```diff
@@ -60,6 +61,8 @@ def make_aug_forward_and_backward(bsym: BoundSymbol) -> tuple[Callable, Callable
         return cached_result
 
     joint_trace = thunder.trace(inline_trace=False, use_dce=False)(joint_forward_backward, *bsym.args, **bsym.kwargs)
+    # dce is necessary to remove duplicated shape queries, otherwise the trace might overwritten NumberProxy variables
+    joint_trace = dce(joint_trace)
```

Comment:
We have a `use_dce` flag that could be used here for some reason. Conveniently, it defaults to `True`, so there's no need even to spell it out here.
```suggestion
    # dce is necessary to remove duplicated shape queries, otherwise the trace might overwritten NumberProxy variables
    joint_trace = thunder.trace(inline_trace=False, use_dce=True)(joint_forward_backward, *bsym.args, **bsym.kwargs)
```

Reply:
Thanks a lot, somehow I totally missed that arg in trace. :bow: